### PR TITLE
Add new class LangevinThermostatOnRadius for implementing hierarchical strategy

### PR DESCRIPTION
**Pull request by hidekb**
For implementing hierarchical strategy of polymer melts, I added new class LangevinThermostatOnRadius that provide the Langevin thermostat for radii of particles.
Test python file is also added under testsuite directory.

### CODE REVIEW

File: src/integrator/LangevinThermostatOnRadius.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		
+		if(exclusions.count((*cit).id()) == 0) {
+		    frictionThermo(*cit);
+		}
+		
+	    }
+	}
+	
+	void LangevinThermostatOnRadius::frictionThermo(Particle& p) {
+	    real massf = sqrt(dampingmass);
```
Comment (MrTheodor) [inline src/integrator/LangevinThermostatOnRadius.cpp]: This seems to be constant and could be set in the constructor already

File: src/integrator/LangevinThermostatOnRadius.hpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	    /** Opposite to heatUp */
+	    void coolDown();
+	    
+	    /** Register this class so it can be used from Python. */
+	    static void registerPython();
+	    
+	private:
+	    
+	    boost::signals2::connection _initialize, _heatUp, _coolDown,
+		_thermalize, _thermalizeAdr;
```
Comment (MrTheodor) [inline src/integrator/LangevinThermostatOnRadius.hpp]: `_thermalizeAdr` is not used anywhere

File: src/integrator/LangevinThermostatOnRadius.cpp [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	void LangevinThermostatOnRadius::thermalize() {
+	    LOG4ESPP_DEBUG(theLogger, "thermalize");
+	    
+	    System& system = getSystemRef();
+	    
+	    CellList cells = system.storage->getRealCells();
+	    
+	    for(CellListIterator cit(cells); !cit.isDone(); ++cit) {
+		
+		if(exclusions.count((*cit).id()) == 0) {
```
Comment (MrTheodor) [inline src/integrator/LangevinThermostatOnRadius.cpp]: maybe `cit->id()`

File: src/integrator/LangevinThermostatOnRadius.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+>>> radius_mass = mass
+>>> # set virtual mass for dynamics of radius
+>>> langevin = espressopp.integrator.LangevinThermostatOnRadius(system, radius_mass)
+>>> # set up the thermostat
+>>> langevin.gamma = gamma
+>>> # set friction coefficient gamma
+>>> langevin.temperature = temp
+>>> # set temperature
+>>> langevin.adress = True
```
Comment (MrTheodor) [inline src/integrator/LangevinThermostatOnRadius.py]: where did you define `adress` property?

File: src/integrator/LangevinThermostatOnRadius.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+
+
+r"""
+********************************************
+**espressopp.integrator.LangevinThermostatOnRadius**
+********************************************
+Langevin Thermostat for Radii of Particles
```
Comment (MrTheodor) [inline src/integrator/LangevinThermostatOnRadius.py]: If you have some reference for this kind of thermostat, please add it.

File: testsuite/langevin_thermostat_on_radius/test_langevin_thermostat_on_radius.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        # integrator on radius
+        radius_mass = 10.
+        integratorOnRadius  = espressopp.integrator.VelocityVerletOnRadius(self.system, dampingmass=radius_mass)
+        integrator.addExtension(integratorOnRadius)
+
+        # Langevin Thermostat on Radius
+        langevin = espressopp.integrator.LangevinThermostatOnRadius(self.system, dampingmass=radius_mass)
+        langevin.gamma = 1.0
+        langevin.temperature = 10.0
+        langevin.adress = False
```
Comment (MrTheodor) [inline testsuite/langevin_thermostat_on_radius/test_langevin_thermostat_on_radius.py]: same here

File: src/integrator/LangevinThermostatOnRadius.cpp (hunk 3 above)
Comment (hidekb) [inline src/integrator/LangevinThermostatOnRadius.cpp]: I corrected (*cit).id() to cit->id().

File: src/integrator/LangevinThermostatOnRadius.cpp (hunk 1 above)
Comment (hidekb) [inline src/integrator/LangevinThermostatOnRadius.cpp]: Yes. Then, I modified that massf is set in the constructor.

File: src/integrator/LangevinThermostatOnRadius.hpp (hunk 2 above)
Comment (hidekb) [inline src/integrator/LangevinThermostatOnRadius.hpp]: I removed `_thermalizeAdr`.

File: src/integrator/LangevinThermostatOnRadius.py (hunk 4 above)
Comment (hidekb) [inline src/integrator/LangevinThermostatOnRadius.py]: Nowhere and this property is unnecessassary.
Hence, I removed this.

File: testsuite/langevin_thermostat_on_radius/test_langevin_thermostat_on_radius.py (hunk 6 above)
Comment (hidekb) [inline testsuite/langevin_thermostat_on_radius/test_langevin_thermostat_on_radius.py]: I removed this.

File: src/integrator/LangevinThermostatOnRadius.py (hunk 5 above)
Comment (hidekb) [inline src/integrator/LangevinThermostatOnRadius.py]: I don't have any reference for this. Actually, this is just normal langevin thermostat that only affect not the position but the radius of particles.
For confirmation, `LangevinThermostat` only affect the position of particles.

**Pull request merged**